src/solvers/refinement/string_refinement.cpp

```diff
@@ -1685,14 +1701,26 @@ exprt string_refinementt::get(const exprt &expr) const
     if(it!=found_length.end())
       return get_array(ecopy, it->second);
   }
+  else if(refined_string_typet::is_refined_string_type(ecopy.type()) &&
```

Comment:
What's the reason for having this special case. Get should already do basically the same thing.
I prefer not making it more complicated.